---
title: "Population genetics analyses compiled"
author: "Julia Harenčár"
date: "5/2/2022"
format: 
  html:
    code-fold: true
    code-tools: true
    code-overflow: wrap
    toc: true
    number-sections: true
theme:
  light: minty
  dark: superhero
editor_options: 
  chunk_output_type: console
---

```{r}
#| echo: false
library(ggplot2)
library(ggrepel)
library(patchwork)
library(dplyr)
library(tidyverse)
```

author: "Julia Harenčár" date: "5/2/2022" output: html_document

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	fig.height = 3,
	fig.path = "ANGSD/PCAngsd_figs/",
	fig.width = 5
)
knitr::opts_knit$set(root.dir = '/Users/Julia/Documents/GitHub/Population_genetics_vill_alle')
#knitr::opts_knit$set(root.dir = '/Users/Julia/Library/CloudStorage/GoogleDrive-jharenca@ucsc.edu/My Drive/GitHub/seq_analysis/ANGSD')
```
## PCAs
Below, I summarize the PCAs I generated by running PCAngsd on the genotype likelihood (beagle) file generated from clean bams with greater than 0.05x coverage WITH -sites specified based on bcftools site filtering.

#### PCA of LGAHVB

generally a 0.15 cutoff

```{r full_angsd1_AVH_PCA, echo=FALSE}
LGAHVBpop <- read.table("ANGSD/angsd_out/LGAHVB_231220.pop.info")
cov <- as.matrix(read.table("ANGSD/angsd_out/PCAngsd/LGAHVB_231220.PCAngsd.cov"))
e <- eigen(cov)

PC1 <- e$vectors[,1]
PC2 <- e$vectors[,2]
spp <- LGAHVBpop[,2]
ID <- LGAHVBpop[,1]
PC12.LGAVHB.co0.15 <- data.frame(PC1, PC2, ID, spp)

# Calculate the proportion of variance explained by each PC
variance_explained <- e$values / sum(e$values)
variance_explained[1] # 0.115
variance_explained[2] # 0.101

ggplot(PC12.LGAVHB.co0.15, aes(x=PC1, y=PC2, fill=spp)) +
  geom_point(size = 3, pch = 21, colour="black") +
  scale_x_reverse() +
  theme_bw() + 
  geom_text_repel(aes(label=ID), size =1.5, nudge_y = 0.03, max.overlaps = 30)
#ggtitle("AVH1.selected.sites.co0.15x") +
#geom_text_repel(aes(label=ID), size =3, nudge_y = 0.03)
 
  
## example of how to add repelled text labels:
#library(ggrepel)
#ggplot(mtcars, aes(wt, mpg, label = rownames(mtcars))) +
#  geom_text_repel() +
#  geom_point(color = 'red') +
#  theme_classic(base_size = 16)
```

#### PCAs of alle, vill, and hybrids only (ANGSD_1; only selected sites with 0.05x coverage cutoff)

PCAngsd run on beagle file from angsd1 (samtools; see 'ANGSD run tracking - full dataset' in `[g-drive]/GitHub/seq_analysis/ANGSD/angsd_runs` with estimation of independent allele frequencies. Plot created from output of the following code:\
`python pcangsd.py -beagle $IN_DIR/selected_sites_0.05co_AVH.genolike.beagle.gz -o $OUT_DIR/sites_0.15co_AVH_2_PCAngsd`

```{r full_angsd1_AVH_PCA, echo=FALSE}
#latest (240121)
#Number of sites after MAF filtering (0.05): 13227
AVHF1pop <- read.table("ANGSD/angsd_out/AHVF1_0.15co_240109.clean.pop.info")
cov <- as.matrix(read.table("ANGSD/angsd_out/PCAngsd/AHVF1_240109.PCAngsd.cov"))
e <- eigen(cov)

PC1 <- e$vectors[,1]
PC2 <- e$vectors[,2]
spp <- AVHF1pop[,2]
ID <- AVHF1pop[,1]
PC12.AVHF1.co0.15 <- data.frame(PC1, PC2, ID, spp)

# Calculate the proportion of variance explained by each PC
variance_explained <- e$values / sum(e$values)
variance_explained[2] # only like 2%, so very little and can ignore

#number of sites in the dataset:
num_sites <- 13227

# With hand pollinated likely backcrosses (vill pollen, hybrid mom)
ggplot(PC12.AVHF1.co0.15, aes(x=PC1, y=PC2, fill=spp)) +
  geom_point(size = 3, pch = 21, colour="black") +
  scale_fill_manual(values=c("#CC79A7", "#93c572","#56B4E9", "#FFD700"), name=paste("Species"), labels=c(expression(italic("C. allenii")), "F1","hybrid", expression(italic("C. villosissimus")))) +
  scale_x_reverse() +
  theme_bw() +
  theme(legend.text.align = 0) + 
  xlab("PC1 ()") +
  ylab("PC2 (1.9%") + 
  labs(x = paste0("PC1 (", round(variance_explained[1], 4)*100, "%)"), y = paste0("PC2 (", round(variance_explained[2], 3)*100, "%)")) +
   annotate("text", x = -0.17, y = 0.25, label = paste("# sites =", num_sites), hjust = 1.1, vjust = -1.5, size = 4, color = "black")
  
# Without hand pollinated likely backcrosses (vill pollen, hybrid mom) #other yellow: #F0E442
n_alle <- PC12.AVHF1.co0.15 %>% filter(spp == "ALLE") %>% nrow()
n_vill <- PC12.AVHF1.co0.15 %>% filter(spp == "VILL") %>% nrow()
n_hyb <- PC12.AVHF1.co0.15 %>% filter(spp == "HYB") %>% nrow()
  
AHV_PCA <- PC12.AVHF1.co0.15 %>% 
  filter(spp !="F1") %>% 
  ggplot(aes(x=PC1, y=PC2, fill=spp)) +
  geom_point(size = 3, pch = 21, colour="black") +
  scale_fill_manual(values=c("#CC79A7", "#56B4E9", "#FFD700"), name=paste("Species"), labels=c(bquote(italic("C. allenii")~"(n="*.(n_alle)*")"), bquote(~"hybrid"~"(n="*.(n_hyb)*")"), bquote(italic("C. villosissimus")~"(n="*.(n_vill)*")"))) +
  theme_bw() +
  theme(legend.text.align = 0) + 
  xlab("PC1 ()") +
  ylab("PC2 (1.9%") + 
  labs(x = paste0("PC1 (", round(variance_explained[1], 3)*100, "%)"), y = paste0("PC2 (", round(variance_explained[2], 2)*100, "%)")) +
   annotate("text", x = 0.18, y = 0.28, label = paste("# sites =", num_sites), hjust = 1.1, vjust = -1.5, size = 3, color = "black") +
  geom_text_repel(aes(label=ID), size =1.5, nudge_y = 0.03, max.overlaps = 30)

ggsave("ANGSD/PCAngsd_figs/AHV_PCA_240121.png", device = "png", width = 15, height = 7, units = "cm")

#  geom_text_repel(aes(label=ID), size =1.5, nudge_y = 0.03, max.overlaps = 30)
#ggtitle("AVH1.selected.sites.co0.15x") +
#geom_text_repel(aes(label=ID), size =3, nudge_y = 0.03)

## older
AVHF1pop <- read.table("ANGSD/angsd_out/AHVF1_231220.pop.info")
cov <- as.matrix(read.table("ANGSD/angsd_out/PCAngsd/AHVF1_231220.PCAngsd.cov"))
e <- eigen(cov)

PC1 <- e$vectors[,1]
PC2 <- e$vectors[,2]
spp <- AVHF1pop[,2]
ID <- AVHF1pop[,1]
PC12.AVHF1.co0.15 <- data.frame(PC1, PC2, ID, spp)

# Calculate the proportion of variance explained by each PC
variance_explained <- e$values / sum(e$values)
variance_explained[2] # only like 2%, so very little and can ignore

ggplot(PC12.AVHF1.co0.15, aes(x=PC1, y=PC2, fill=spp)) +
  geom_point(size = 3, pch = 21, colour="black") +
  scale_fill_manual(values=c("#CC79A7", "#93c572","#56B4E9", "#FFD700"), name=paste("Species"), labels=c(expression(italic("C. allenii")), "F1","hybrid", expression(italic("C. villosissimus")))) +
  scale_x_reverse() +
  theme_bw() +
  theme(legend.text.align = 0) + 
  geom_text_repel(aes(label=ID), size =1.5, nudge_y = 0.03, max.overlaps = 30)
#ggtitle("AVH1.selected.sites.co0.15x") +
#geom_text_repel(aes(label=ID), size =3, nudge_y = 0.03)
 
  
## example of how to add repelled text labels:
#library(ggrepel)
#ggplot(mtcars, aes(wt, mpg, label = rownames(mtcars))) +
#  geom_text_repel() +
#  geom_point(color = 'red') +
#  theme_classic(base_size = 16)
```

#### PCAs of alle, vill, hybrids, guan, and laev (ANGSD_1; only clean bams with 0.15x coverage cutoff - less than 35bp and lower than 20 mapping Q removed)

##FIX## #PCAngsd run on beagle file from angsd1 (samtools; see 'ANGSD run tracking - full dataset' in \#`[g-drive]/GitHub/seq_analysis/ANGSD/angsd_runs` with estimation of independent allele frequencies. #Plot created from output of the following code:\
\#`pcangsd.py -beagle $IN_DIR/full_AVHLG_genolike.beagle.gz -o $OUT_DIR/AVHLG_1_PCAngsd -threads 44`

```{r full_angsd1_AVHLG_PCA, echo=FALSE}
LGVHApop <- read.table("ANGSD/angsd_out/LGVHA.clean.pop.info")
cov <- as.matrix(read.table("ANGSD/angsd_out/PCAngsd/sites_0.15co_nosml_q20_LGVHA_PCAngsd.cov"))
e <- eigen(cov)

PC1 <- e$vectors[,1]
PC2 <- e$vectors[,2]
spp <- LGVHApop[,2]
ID <- LGVHApop[,1]
PC12.LGVHA.co0.15X <- data.frame(PC1, PC2, ID, spp)

library(ggplot2)
ggplot(PC12.LGVHA.co0.15X, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp))) +
  ggtitle("LGVHA.co0.15x") +
  geom_text_repel(aes(label=ID), size =3, nudge_y = 0.03) 
 
  

#labeling code to check out:
#ggplot(tab, aes(x = EV2, y = EV1, col = pop)) +
#  geom_point(alpha = 0.5, size = 5) +
#  # geom_text(aes(label=sample.id), size =3, hjust = 0, nudge_x = 0.02) +
#  # geom_text(aes(label=sample.id), size =3, position=position_jitter(width=0.2,height=0,2)) +
#  geom_text_repel(aes(label=sample.id), size =3, nudge_y = 0.03) +
#  theme_bw()
```

## ABBA/BABA

```{r}
library('pracma')
```

### Whole genome

Code to run abba baba test and calculate D-statistic with angsd:

```{bash}
# code to run abba/baba test
angsd -doAbbababa2 1 -bam BAV_clean_ordered_0.3co_240219.bam.filelist -sizeFile sizeFile.size -doCounts 1 -out lasiOUT_full_240219 -anc /hb/groups/kay_lab/genomes/C.lasius_genome_NCBI.fasta -minQ 30 -minMapQ 30 -enhance 1 -only_proper_pairs 1 -baq 1 -ref /hb/groups/kay_lab/genomes/C.lasius_genome_NCBI.fasta -blockSize 5000000 -nThreads 96

# code to calculate D-statistic and associated stats
cd ANGSD/angsd_out/doAbbababa2
Rscript estAvgError.R angsdFile="lasiOUT_full_240215" out="lasiOUT_full_240215" sizeFile=sizeFile.size nameFile=popNames.name

# --- Table of Results --- Combination  1  out of  3  ---
# --- H1= bracteatus  H2= allenii  H3= villosissimus  H4= lasius  ---
# ---------------------------------------------------------------------------------
#   Mode          |Dstat          |sd(Dstat)      |Djack          |Zscore |Pvalue
# ---------------------------------------------------------------------------------
# Observed        |1.141e-01      |1.194e-02      |1.141e-01      |9.557  |0.0e+00
# ---------------------------------------------------------------------------------
# No Trans        |1.665e-01      |1.533e-02      |1.665e-01      |10.861 |0.0e+00
# ---------------------------------------------------------------------------------
```

### Per Chromosome

Code to run abba baba test and calculate D-statistic with angsd:

```{bash}
# use array to run eparately on each of 9 chromosomes:
angsd -doAbbababa2 1 -bam BAV_clean_ordered_0.3co_240219.bam.filelist -sizeFile sizeFile.size -doCounts 1 -out lasiOUT_chr${SLURM_ARRAY_TASK_ID}_240219 -anc /hb/groups/kay_lab/genomes/C.lasius_genome_NCBI.fasta -minQ 30 -minMapQ 30 -only_proper_pairs 1 -baq 1 -ref /hb/groups/kay_lab/genomes/C.lasius_genome_NCBI.fasta -blockSize 5000000 -nThreads 96 -r Chrom${SLURM_ARRAY_TASK_ID}

# use loop to calculate D-stats with R script for each:
for i in {1..9}; do
        Rscript estAvgError.R angsdFile="lasiOUT_chr${i}_240219" out="lasiOUT_chr${i}_240219" sizeFile=sizeFile.size nameFile=popNames.name
done
```

```{r}
# Per chromosome  values and p-val correction
raw_pvals <- c(0.000079, 0.027063, 0.000168, 0.015394, 0.040503, 0.000393, 0.04996, 0.000095, 0.152879)
FDR_pvals <- p.adjust(raw_pvals, method = "BH") # I notice that the highest p-val doesn't change at all... 
FDR_pvals # 0.00042750 0.04059450 0.00050400 0.02770920 0.05207529 0.00088425 0.05620500 0.00042750 0.15287900
```

## Dsuite

### Per Chromosome Dstats and F4-ratio

```{r}
# Per chromosome  values and p-val correction
raw_pvals <- c(8.12e-07, 2.66e-06, 4.52e-13,  1.39e-10, 7.16e-09, 1.27e-07, 1.20e-06, 1.05e-10, 6.95e-05)
FDR_pvals <- p.adjust(raw_pvals, method = "BH")
```

### Plotting f_dM

#### 50 variant windows, 25 variant steps

```{r}
# Read all chromosome files intoo single dataframe
# Create an empty dataframe to store the combined data
LAEVout_w50s25_all9 <- data.frame()

# List all the .tsv files in the directory
file_list <- list.files(path = "./Dsuite/f_dM_w50s25/", pattern = "*.tsv", full.names = TRUE)

# Loop through each file and read it into a temporary dataframe
for (file in file_list) {
  temp_df <- read.delim(file, sep = "\t")
  
  # Combine the temporary dataframe with the existing data
  LAEVout_w50s25_all9 <- rbind(LAEVout_w50s25_all9, temp_df)
}

# Plotting 50,25 f_dM

# remove small contigs, keep only chromosomes
LAEVout_w50s25_all9 <- LAEVout_w50s25_all9 %>% 
  filter(startsWith(chr, "Chrom"))

# check for failed rows
# LAEVout_w50s25_all9[is.na(LAEVout_w50s25_all9$f_dM), ]
# remove failed rows
LAEVout_w50s25_all9 <- LAEVout_w50s25_all9[!is.na(LAEVout_w50s25_all9$f_dM), ]

# create thresholds of mean + and - three standard deviations
mean_plus_three_sd <- mean(LAEVout_w50s25_all9$f_dM)+3*sd(LAEVout_w50s25_all9$f_dM)
mean_minus_three_sd <- mean(LAEVout_w50s25_all9$f_dM)-3*sd(LAEVout_w50s25_all9$f_dM)

# create single bp location for x-axis that adds previous chromosmes lengths
data_cum <- LAEVout_w50s25_all9 %>% 
  group_by(chr) %>% 
  summarise(max_windowStart = max(windowStart)) %>% 
  mutate(windowStart_add = lag(cumsum(max_windowStart), default = 0)) %>% 
  select(chr, windowStart_add)

LAEVout_w50s25_all9 <- LAEVout_w50s25_all9 %>% 
  inner_join(data_cum, by = "chr") %>% 
  mutate(windowStart_cum = windowStart + windowStart_add)

axis_set <- LAEVout_w50s25_all9 %>% 
  group_by(chr) %>% 
  summarize(center = mean(windowStart_cum))

ggplot(data=LAEVout_w50s25_all9, aes(x=windowStart_cum, y=f_dM, color = as_factor(chr))) +
  geom_point() +
  geom_hline(yintercept = c(mean_plus_three_sd, mean_minus_three_sd), linetype = "dashed", color = "red") +
  scale_color_manual(values = rep(c("black", "darkgray"), unique(length(axis_set$chr)))) +
  guides(color = "none") +
  geom_point(data = subset(LAEVout_w50s25_all9, f_dM > mean_plus_three_sd | f_dM < mean_minus_three_sd), color = "red") +
  scale_x_continuous(label = axis_set$chr, breaks = axis_set$center) +
  labs(x = NULL, 
       y = "f<sub>dM")

# Save plot
ggsave("Dsuite/LAEVout_w50s25_f_dM_allChrs.png", device = "png", width = 110, height = 7, units = "cm")

```
